packages/react-avatar/src/components/Avatar/useAvatar.tsx

```diff
@@ -53,6 +58,41 @@ export const useAvatar = (props: AvatarProps, ref: React.Ref<HTMLElement>, defau
   return state;
 };
 
+/**
+ * Avatar treats shorthand for the image and badge props differently. Rather than the string being
+ * the child of those slots, they translate to the image's src and the badge's status prop.
+ */
+const resolveAvatarShorthandProps = (props: AvatarProps) => {
+  if (typeof props.image === 'string') {
+    props = { ...props, image: { src: props.image, children: null } };
+  }
+
+  if (typeof props.badge === 'string') {
+    props = { ...props, badge: { status: props.badge, children: null } };
+  }
```

Comment:
Just a recommendation, you are potentially spreading `props` twice here which is expensive, it might be better to do something like this to achieve the same result and only spread `props` once:

```suggestion
  if (typeof props.image === 'string' || typeof props.badge === 'string') {
    props = {
      ...props,
      image: typeof props.image === 'string' ? { src: props,image, children: null } : props.image,
      badge: typeof props.badge === 'string' ? { status: props,badge, children: null } : props.badge,
    }
  }
```